core/src/main/java/org/kiwix/kiwixmobile/core/utils/ServerUtils.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.utils
+
+import java.net.InetAddress
+import java.net.NetworkInterface
+import java.net.SocketException
+
+object ServerUtils {
+  @JvmField var port = 0
+  @JvmField var isServerStarted = false
+  const val INVALID_IP = "-1" // To remove extra characters from IP for Android Pie
+
+  // get Ip address of the device's wireless access point i.e. wifi hotspot OR wifi network
+  @Suppress("MagicNumber")
+  @JvmStatic fun getIpAddress(): String? {
+    var ip = ""
+    try {
+      val enumNetworkInterfaces = NetworkInterface.getNetworkInterfaces()
```

Comment:
this whole section can be improved a lot just even by using for loops

Reply:
there is a solution with `map/filter/fold/flatmap` which is much cooler than using the for loops, definitely take a stab at it